packages/notification-service/src/firebase.ts

```diff
@@ -122,29 +104,44 @@ export function initializeDb() {
       rewardsSenders = snapshot?.val() ?? []
       console.debug('Rewards senders updated: ', rewardsSenders)
     },
-    (errorObject: any) => {
-      console.error('Rewards senders data read failed:', errorObject.code)
-    }
+    firebaseFetchError('Rewards senders')
   )
 
-  database.ref('/inviteRewardAddresses').on('value', (snapshot) => {
-    const addresses = snapshot?.val() ?? []
-    console.debug(`inviteRewardAddresses fetched: ${addresses}`)
-    if (addresses.length) {
-      inviteRewardsSenders = addresses
-    }
-  })
+  database.ref('/inviteRewardAddresses').on(
+    'value',
+    (snapshot) => {
+      const addresses = snapshot?.val() ?? []
+      console.debug(`inviteRewardAddresses fetched: ${addresses}`)
+      if (addresses.length) {
+        inviteRewardsSenders = addresses
+      }
+    },
+    firebaseFetchError('Invite rewards addresses')
+  )
 
   knownAddressesCache = new KnownAddressesCache()
   knownAddressesCache.startListening(database)
 }
 
-export function getTokenFromAddress(address: string) {
-  return registrations[address]?.fcmToken ?? null
+export async function getRegistration(address: string) {
+  let registration: Registration | undefined
+  await registrationsRef.once(
+    'value',
+    (snapshot) => {
+      registration = (snapshot?.val() || {})[address]
+    },
+    firebaseFetchError('Registration')
+  )
+  return registration
+}
+
+export async function getTokenFromAddress(address: string) {
```

Comment:
Doing this this way means making two requests for each transaction to send: one for fetching the language and another for fetching the fcmToken. I suggest instead fetching the registration node for an address only once on both the `sendPaymentNotification` and `sendInviteNotification` functions and adding a parameter to `sendNotification` called `fcmToken` to pass it directly we that we can avoid this :)